clients/include/testing_gemv_batched.hpp

```diff
@@ -65,121 +61,111 @@ hipblasStatus_t testing_gemvBatched(const Arguments& argus)
-    err_y = hipMemcpy(dy_array, by_array, batch_count * sizeof(T*), hipMemcpyHostToDevice);
-    if(err_A != hipSuccess || err_x != hipSuccess || err_y != hipSuccess)
-    {
-        return HIPBLAS_STATUS_MAPPING_ERROR;
-    }
+    CHECK_HIP_ERROR(dA.transfer_from(hA));
+    CHECK_HIP_ERROR(dx.transfer_from(hx));
+    CHECK_HIP_ERROR(dy.transfer_from(hy));
+    CHECK_HIP_ERROR(hipMemcpy(d_alpha, &h_alpha, sizeof(T), hipMemcpyHostToDevice));
+    CHECK_HIP_ERROR(hipMemcpy(d_beta, &h_beta, sizeof(T), hipMemcpyHostToDevice));
 
     /* =====================================================================
            HIPBLAS
     =================================================================== */
     if(argus.unit_check || argus.norm_check)
     {
+        CHECK_HIPBLAS_ERROR(hipblasSetPointerMode(handle, HIPBLAS_POINTER_MODE_HOST));
+        CHECK_HIPBLAS_ERROR(hipblasGemvBatchedFn(handle,
+                                                 transA,
+                                                 M,
+                                                 N,
+                                                 (T*)&h_alpha,
+                                                 dA.ptr_on_device(),
+                                                 lda,
+                                                 dx.ptr_on_device(),
+                                                 incx,
+                                                 (T*)&h_beta,
+                                                 dy.ptr_on_device(),
+                                                 incy,
+                                                 batch_count));
+
+        CHECK_HIP_ERROR(hy_host.transfer_from(dy));
+        CHECK_HIP_ERROR(dy.transfer_from(hy));
+
+        CHECK_HIPBLAS_ERROR(hipblasSetPointerMode(handle, HIPBLAS_POINTER_MODE_DEVICE));
         CHECK_HIPBLAS_ERROR(hipblasGemvBatchedFn(handle,
                                                  transA,
                                                  M,
                                                  N,
-                                                 (T*)&alpha,
-                                                 dA_array,
+                                                 d_alpha,
+                                                 dA.ptr_on_device(),
                                                  lda,
-                                                 dx_array,
+                                                 dx.ptr_on_device(),
                                                  incx,
-                                                 (T*)&beta,
-                                                 dy_array,
+                                                 d_beta,
+                                                 dy.ptr_on_device(),
                                                  incy,
                                                  batch_count));
 
+        CHECK_HIP_ERROR(hy_device.transfer_from(dy));
```

Comment:
Could be pushed after the cpu reference to hide some of the cpu ref. calculation time.